data_model/src/transaction.rs

```diff
@@ -395,6 +395,13 @@ impl AsRef<[u8]> for WasmSmartContract {
     }
 }
 
+impl WasmSmartContract {
+    /// Create [`Self`] from raw wasm bytes
+    pub fn from_raw(blob: Vec<u8>) -> Self {
+        Self(blob)
+    }
+}
```

Comment:
there is `WasmSmartContract::new`. Remove either one

Reply:
This needs to be `#[inline]` for move elision to work properly. Also neither of the two methods are very informative. `from_compiled` sounds a lot better than `from_raw` because raw here could mean 
1. The source code of the Verifier, 
2. The results of compiling with the WASM target
3. The results of `wasmtime` JIT.

Reply:
In general given that we'd be using WASM for a lot more than just on-chain scripting, we might want to consider making the object hierarchy more pronounced: example, `wasm::SmartContract` and `wasm::Verifier` being two newtype wrappers around blobs, which however, `impl trait wasm::Executable`. Extract common logic into extension traits.

Reply:
I've extracted your comment into separate ticket to further discussion (#3443).